### PR TITLE
sync/ErrorWaiter: tests

### PR DESCRIPTION
**Pull request by abhinav**

### CODE REVIEW

**Comment by abhinav** [comment]
@yarpc/golang

File: sync/err_test.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+			"multiple errors",
+			[]error{nil, errors.New("1"), errors.New("2"), nil},
+			[]error{errors.New("1"), errors.New("2")},
+		},
+	}
+
+	for _, tt := range tests {
+		var ew ErrorWaiter
+		for _, err := range tt.errs {
+			// Need to create a function because the 'err' changes
```
Comment (prashantv) [inline sync/err_test.go]: this is fine, but another way is,
```go
errLocal := err
ew.Subimt(func() error { return errLocal })
```

File: sync/err_test.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	"github.com/stretchr/testify/assert"
+)
+
+func TestErrorWaiter(t *testing.T) {
+	tests := []struct {
+		desc string
+		errs []error
+		want []error
+	}{
+		{
```
Comment (prashantv) [inline sync/err_test.go]: can you also add a test for empty list (just `nil` or `[]error{}`) .. or both

**Comment by prashantv** [comment]
lgtm

**Comment by breerly** [comment]
:shipit:

File: sync/err_test.go (hunk 1 above)
Comment (abhinav) [inline sync/err_test.go]: 👍

File: sync/err_test.go (hunk 2 above)
Comment (abhinav) [inline sync/err_test.go]: 👍
